### DISCUSSION TITLE
Swipe Left/Right gestures TabBarItems

### DISCUSSION THREAD

**Discussion by jindal1979** [unoplatform/uno#11405] [Q&A]
I am trying to create Top TabBar with a few TabBarItems. I want to select the TabBarItem by using Swipe Left/Right gestures. Is there an API to enable swipe left/right gestures in TabBar or any other recommended approach? Please advise.

**Comment by jeromelaban** [comment]
Thanks for the question. Can you post a sample of the code you trying to get working?

**Reply by jindal1979** [reply]
I don't have a code, I want to implement it and therefore asking for directions.

**Reply by jeromelaban** [reply]
Thanks. I'm not sure to see which pattern you want to implement. Would you have an example of an app which does this? If you can share a video, it'll be easier.

**Comment by jindal1979** [comment]
[![image](https://user-images.githubusercontent.com/22019325/220542909-5eea7096-a6e9-4967-a099-e5511715fe77.png)

In this image, there are 4 tabs in Tabbar. The current position is 3, if I swipe left the Tabar with object 2 should be selected or if I swipe right, the Tabbar with object 4 should be selected as the current TabbarItem.

**Reply by jeromelaban** [reply]
@kazo0

**Reply by Xiaoy312** [reply]
do you intend to swipe in the tabbar area "object 1, obj 2..." or in the content area "3"?
if it is for the content area, you should be able to use a FlipView and chain its SelectedIndex to that of TabBar

**Reply by kazo0** [reply]
If you are indeed looking to be able to swipe the content area, you can use something like a FlipView that was mentioned above and you can synchronize the selected item by using the TabBarSelectorBehavior from Toolkit. You can use the [Selector attached property](https://github.com/unoplatform/uno.toolkit.ui/blob/main/src/Uno.Toolkit.UI/Behaviors/TabBarSelectorBehavior.Properties.cs#L38) and it will automatically sync the TabBar's selected item to whatever Selector you provided.

**Reply by jindal1979** [reply]
> Thanks for suggesting FlipView. However, I have four different Pages and I use Frame to navigate between these 4 pages based on TabBar SelectionChanged event. If I use FlipView all 4 pages will be loaded when FlipView is loaded, I want to avoid it.

**Comment by jindal1979** [comment]
Thanks for suggesting FlipView. However, I have four different Pages and I use Frame to navigate between these 4 pages based on TabBar SelectionChanged event.  If I use FlipView all 4 pages will be loaded when FlipView is loaded, I want to avoid it.

**Reply by Xiaoy312** [reply]
You can try to bind each page's [`x:Load`](https://learn.microsoft.com/en-us/windows/uwp/xaml-platform/x-load-attribute) to its selection state for dynamic loading and unloading.

**Reply by jindal1979** [reply]
Thanks !!
